pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -1035,3 +1030,25 @@ func getConfigDataFromHubURL(th *v1alpha1.TektonHub) (*Data, error) {
 
 	return data, nil
 }
+
+// Creates hub target namespace, if it's not found
+func (r *Reconciler) createNameSpace(ctx context.Context, th *v1alpha1.TektonHub) error {
+	_, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, th.Spec.GetTargetNamespace(), metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			ownerRef := *metav1.NewControllerRef(th, th.GroupVersionKind())
+
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            th.Spec.TargetNamespace,
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
```

Comment:
I have a little concern on owner reference.

There is a possibility: 
* `hub` installed earlier than `pipeline` on `tekton-pipelines` namespace
* later `pipeline` installed on `tekton-pipelines` namespace (still the owner reference will be `hub`)
* now, if we delete the `hub` CR, that will remove the `tekton-pipelines` namespace along with `pipeline`

@piyush-garg @PuneetPunamiya

Reply:
Yes namesapce will be deleted and recreated